closed/autoconf/custom-hook.m4

```diff
@@ -360,52 +374,56 @@ AC_DEFUN([OPENJ9_PLATFORM_SETUP],
       OPENJ9_PLATFORM_CODE=xa64
     elif test "x$OPENJDK_BUILD_OS" = xwindows ; then
       OPENJ9_PLATFORM_CODE=wa64
-      if test "x$OPENJ9_LIBS_SUBDIR" = xdefault ; then
-        OPENJ9_BUILDSPEC=win_x86-64
-      else
-        OPENJ9_BUILDSPEC=win_x86-64_cmprssptrs
-      fi
+      OPENJ9_BUILD_OS=win
     elif test "x$OPENJDK_BUILD_OS" = xmacosx ; then
       OPENJ9_PLATFORM_CODE=oa64
-      if test "x$OPENJ9_LIBS_SUBDIR" = xdefault ; then
-        OPENJ9_BUILDSPEC=osx_x86-64
-      else
-        OPENJ9_BUILDSPEC=osx_x86-64_cmprssptrs
-      fi
+      OPENJ9_BUILD_OS=osx
     else
       AC_MSG_ERROR([Unsupported OpenJ9 platform ${OPENJDK_BUILD_OS}!])
     fi
   elif test "x$OPENJ9_CPU" = xppc-64_le ; then
     OPENJ9_PLATFORM_CODE=xl64
-    if test "x$OPENJ9_LIBS_SUBDIR" != xdefault ; then
-      OPENJ9_BUILDSPEC="${OPENJDK_BUILD_OS}_ppc-64_cmprssptrs_le"
+    if test "x$OPENJ9_ENABLE_MIXED_REFERENCES" = xfalse ; then
+      if test "x$OPENJ9_LIBS_SUBDIR" != xdefault ; then
+        OPENJ9_BUILD_MODE_ARCH="ppc-64_cmprssptrs_le"
+      fi
+    else
+      OPENJ9_BUILD_MODE_ARCH="ppc-64_mxdptrs_le"
     fi
   elif test "x$OPENJ9_CPU" = x390-64 ; then
     OPENJ9_PLATFORM_CODE=xz64
   elif test "x$OPENJ9_CPU" = xppc-64 ; then
     OPENJ9_PLATFORM_CODE=ap64
   elif test "x$OPENJ9_CPU" = xarm ; then
     OPENJ9_PLATFORM_CODE=xr32
-    OPENJ9_BUILDSPEC=linux_arm_linaro
+    OPENJ9_BUILD_OS=linux
+    OPENJ9_BUILD_MODE_ARCH=arm_linaro
     OPENJ9_LIBS_SUBDIR=default
   elif test "x$OPENJ9_CPU" = xaarch64 ; then
     OPENJ9_PLATFORM_CODE=xr64
     if test "x$COMPILE_TYPE" = xcross ; then
-      OPENJ9_BUILDSPEC="${OPENJ9_BUILDSPEC}_cross"
+      OPENJ9_BUILD_MODE_ARCH="${OPENJ9_BUILD_MODE_ARCH}_cross"
     fi
   elif test "x$OPENJ9_CPU" = xriscv64 ; then
     OPENJ9_PLATFORM_CODE=rv64
     if test "x$COMPILE_TYPE" = xcross ; then
-      OPENJ9_BUILDSPEC="${OPENJ9_BUILDSPEC}_cross"
+      OPENJ9_BUILD_MODE_ARCH="${OPENJ9_BUILD_MODE_ARCH}_cross"
     fi
   else
     AC_MSG_ERROR([Unsupported OpenJ9 cpu ${OPENJ9_CPU}!])
   fi
 
+  if test "x$OPENJ9_BUILD_OS" != x ; then
+    OPENJ9_BUILDSPEC="${OPENJ9_BUILD_OS}_${OPENJ9_BUILD_MODE_ARCH}"
+  else
+    OPENJ9_BUILDSPEC="${OPENJDK_BUILD_OS}_${OPENJ9_BUILD_MODE_ARCH}"
+  fi
```

Comment:
If you assign the default above (`OPENJ9_BUILD_OS=$OPENJDK_BUILD_OS`) then this can be unconditional:
```
  OPENJ9_BUILDSPEC="${OPENJ9_BUILD_OS}_${OPENJ9_BUILD_MODE_ARCH}"
```